motionplan/rrtStarConnect.go

```diff
@@ -96,47 +107,21 @@ func (mp *rrtStarConnectMotionPlanner) rrtBackgroundRunner(ctx context.Context,
 	defer close(rrt.solutionChan)
 
 	// setup planner options
-	if rrt.planOpts == nil {
-		rrt.planOpts = newBasicPlannerOptions()
-	}
-	algOpts, err := newRRTStarConnectOptions(rrt.planOpts)
-	if err != nil {
-		rrt.solutionChan <- &rrtPlanReturn{planerr: err}
-		return
+	if mp.planOpts == nil {
+		mp.planOpts = newBasicPlannerOptions()
 	}
 
 	mp.start = time.Now()
 
-	// get many potential end goals from IK solver
-	solutions, err := getSolutions(ctx, rrt.planOpts, mp.solver, goal, seed, mp.Frame(), mp.randseed.Int())
-	mp.logger.Debugf("RRT* found %d IK solutions", len(solutions))
-	if err != nil {
-		rrt.solutionChan <- &rrtPlanReturn{planerr: err}
-		return
-	}
-
-	// publish endpoint of plan if it is known
-	if rrt.planOpts.MaxSolutions == 1 && rrt.endpointPreview != nil {
-		mp.logger.Debug("RRT* found early final solution")
-		rrt.endpointPreview <- solutions[0]
-	}
-
-	// the smallest interpolated distance between the start and end input represents a lower bound on cost
-	_, optimalCost := rrt.planOpts.DistanceFunc(&ConstraintInput{StartInput: seed, EndInput: solutions[0].Q()})
-
-	// initialize maps
-	for i, solution := range solutions {
-		if i == 0 && mp.checkPath(rrt.planOpts, seed, solution.Q()) {
-			rrt.solutionChan <- &rrtPlanReturn{steps: []node{&basicNode{q: seed}, solution}}
-			mp.logger.Debug("RRT* could interpolate directly to goal")
+	if rrt.rm == nil || len(rrt.rm.goalMap) == 0 {
+		planSeed := initRRTMaps(ctx, mp, goal, seed)
+		if planSeed.planerr != nil || planSeed.steps != nil {
+			rrt.solutionChan <- planSeed
 			return
 		}
-		rrt.rm.goalMap[newCostNode(solution.Q(), 0)] = nil
+		rrt.rm = planSeed.rm
 	}
-	mp.logger.Debugf("RRT* failed to directly interpolate from %v to %v", seed, solutions[0].Q())
-	rrt.rm.startMap[newCostNode(seed, 0)] = nil
-
-	target := referenceframe.RandomFrameInputs(mp.frame, mp.randseed)
+	target := referenceframe.InterpolateInputs(seed, rrt.rm.optNode.Q(), 0.5)
```

Comment:
no reason to try this anymore if were already trying to interpolate somewhere else

Reply:
Disagree; if one of the ends is able to reach this point, then even if the other isn't, we have a higher-quality tree than we would otherwise and are somewhat more likely to have a higher-quality output path.

It's not guaranteed to be better, sure, but it's not very expensive either, as it's a single iteration.

Reply:
Ok thats good with me

Reply:
My bad picture tries to illustrate this.  Anyway, this is super minor so I don't really care one way or another, so whatever you decide is good with me

Reply:
Here's a better picture:
![IMG_20221130_134937](https://user-images.githubusercontent.com/18296610/204915542-4560bea6-812d-4d9b-8b6c-5abcec42d871.jpg)

Basically, this will minorly incentivize the green path over the pink one.

This will give the user the impression of "the arm is actively routing around this obstacle" IMO, and will mildly make more sense to an observer.

But agree this is super minor and will almost never matter.

Reply:
Yeah ok lets keep it in then